### DISCUSSION TITLE
Generic parameter 'Result' could not be inferred

### DISCUSSION THREAD

**Discussion by kgrigsby59** [pointfreeco/swift-issue-reporting#28] [Q&A] [answered]
I have the following: 
```
public struct SpotlightClient {
    public var refreshSpotlightIndex: () async throws -> Void

    public init(refreshSpotlightIndex: @escaping () async throws -> Void) {
        self.refreshSpotlightIndex = refreshSpotlightIndex
    }
}

#if DEBUG
import XCTestDynamicOverlay

extension SpotlightClient {
    public static let unimplemented = Self(
        refreshSpotlightIndex: { XCTUnimplemented("\(Self.self).refreshSpotlightIndex") }
    )
}
#endif
```

I'm getting a compile error ```Generic parameter 'Result' could not be inferred``` on the line with XCTUnimplemented.
Am I doing something wrong or is this not implemented for a function that returns Void?

**Comment by stephencelis** [comment, marked as answer]
@kgrigsby59 `XCTUnimplemented` is meant as a dependency "generator" of sorts, so it's responsible for emitting a function that will fail when called later (as opposed to `XCTFail`, which fails immediately when called). I think you can do the following to get things building:

```diff
 extension SpotlightClient {
     public static let unimplemented = Self(
-        refreshSpotlightIndex: { XCTUnimplemented("\(Self.self).refreshSpotlightIndex") }
+        refreshSpotlightIndex: XCTUnimplemented("\(Self.self).refreshSpotlightIndex")
     )
 }
```

Basically, call `XCTUnimplemented` in place of the endpoint's closure, not _within_ the endpoint's closure.

**Reply by kgrigsby59** [reply]
🤦‍♂️
